src/wasm2js.h

```diff
@@ -1766,11 +1766,27 @@ Ref Wasm2JSBuilder::processFunctionBody(Module* m,
               ret = ValueBuilder::makeBinary(left, PLUS, right);
               break;
             case SubInt32:
-              ret = ValueBuilder::makeBinary(left, MINUS, right);
+              if (left->isNumber() && left->getNumber() == 0) {
+                // special case
+                // 0 - x  =>  -x
+                ret = ValueBuilder::makeUnary(MINUS, right);
+              } else {
+                ret = ValueBuilder::makeBinary(left, MINUS, right);
+              }
```

Comment:
It would be good to not do these optimizations when not optimizing. Like we don't run the extra opts-for-js pass when not optimizing.

Reply:
Done!